demos/defender/defender_demo/defender_demo.c

```diff
@@ -0,0 +1,575 @@
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool unsubscribeFromDefenderTopics( void )
+{
+    bool status = true;
+
+    status = UnsubscribeFromTopic( DEFENDER_API_JSON_ACCEPTED( THING_NAME ),
+                                   DEFENDER_API_LENGTH_JSON_ACCEPTED( THING_NAME_LENGTH ) );
+
+    if( status == true )
+    {
+        status = UnsubscribeFromTopic( DEFENDER_API_JSON_REJECTED( THING_NAME ),
+                                       DEFENDER_API_LENGTH_JSON_REJECTED( THING_NAME_LENGTH ) );
+    }
+
+    return status;
+}
+/*-----------------------------------------------------------*/
+
+static bool publishDeviceMetricsReport( uint32_t reportLength )
+{
+    return PublishToTopic( DEFENDER_API_JSON_PUBLISH( THING_NAME ),
+                           DEFENDER_API_LENGTH_JSON_PUBLISH( THING_NAME_LENGTH ),
+                           &( deviceMetricsJsonReport[ 0 ] ),
+                           reportLength );
+}
+/*-----------------------------------------------------------*/
+
+int main( int argc,
+          char ** argv )
+{
+    bool status = true;
+    int exitStatus = EXIT_FAILURE;
+    uint32_t reportLength = 0, i, mqttSessionEstablished = 0;
+
+    /* Silence compiler warnings about unused variables. */
+    ( void ) argc;
+    ( void ) argv;
+
+    /* Start with report not received. */
+    reportStatus = ReportStatusNotReceived;
+
+    /* Set a report Id to be used. */
+    reportId = 1;
+
+    LogInfo( ( "Establishing MQTT session..." ) );
+    status = EstablishMqttSession( publishCallback );
+
+    if( status != true )
+    {
+        LogError( ( "Failed to establish MQTT session." ) );
+    }
+    else
+    {
+        mqttSessionEstablished = 1;
+    }
+
+    if( status == true )
+    {
+        LogInfo( ( "Subscribing to defender topics..." ) );
+        status = subscribeToDefenderTopics();
+
+        if( status != true )
+        {
+            LogError( ( "Failed to subscribe to defender topics." ) );
+        }
+    }
+
+    if( status == true )
+    {
+        LogInfo( ( "Collecting device metrics..." ) );
+        status = collectDeviceMetrics();
+
+        if( status != true )
+        {
+            LogError( ( "Failed to collect device metrics." ) );
+        }
+    }
+
+    if( status == true )
+    {
+        LogInfo( ( "Generating device defender report..." ) );
+        status = generateDeviceMetricsReport( &( reportLength ) );
+
+        if( status != true )
+        {
+            LogError( ( "Failed to generate device defender report." ) );
+        }
+    }
+
+    if( status == true )
+    {
+        LogInfo( ( "Publishing device defender report..." ) );
+        status = publishDeviceMetricsReport( reportLength );
+
+        if( status != true )
+        {
+            LogError( ( "Failed to publish device defender report." ) );
+        }
+    }
+
+    if( status == true )
+    {
+        for( i = 0; i < 5; i++ )
```

Comment:
Let's make a `#define` for the magic number `5`.  Also, is 5 seconds a reliable amount of time to wait for the report?

Reply:
Changed to define and reduced to 2.